go/src/koding/kites/kloud/kloud/kloud.go

```diff
@@ -235,12 +237,45 @@ func New(conf *Config) (*Kloud, error) {
 		},
 	}
 
+	authFn := func(opts *socialapi.AuthOptions) (*socialapi.Session, error) {
+		s, err := modelhelper.FetchOrCreateSession(opts.Session.Username, opts.Session.Team)
+		if err != nil {
+			return nil, err
+		}
+
+		return &socialapi.Session{
+			Username: s.Username,
+			Team:     s.GroupName,
+			ClientID: s.ClientId,
+		}, nil
+	}
+
+	transport := &socialapi.Transport{
+		RoundTripper: storeOpts.Client.Transport,
+		AuthFunc:     socialapi.NewCache(authFn).Auth,
+	}
+
 	kloud.Stack.DescribeFunc = provider.Desc
 	kloud.Stack.CredClient = credential.NewClient(storeOpts)
 	kloud.Stack.MachineClient = machine.NewClient(machine.NewMongoDatabase())
 	kloud.Stack.RemoteClient = &remoteapi.Client{
-		Endpoint: remoteURL,
-		Client:   storeOpts.Client,
+		Client:    storeOpts.Client,
+		Transport: transport,
+	}
+
+	if remoteURL != nil {
+		transport.Host = remoteURL.Host
+
+		localRemoteURL := *remoteURL
+
+		// TODO(rjeczalik): Add PrivateRemoteURL to koding/config.
+		if _, port, err := net.SplitHostPort(localRemoteURL.Host); err == nil {
+			localRemoteURL.Host = net.JoinHostPort("127.0.0.1", port)
+		} else {
+			localRemoteURL.Host = "127.0.0.1"
```

Comment:
Done in https://github.com/koding/koding/pull/9870/commits/5e4f7a6c88f28b9bea1222e204ffeb719e410ebb.